### PR TITLE
(GH-1817) Trigger a GH Action once published

### LINKED ISSUE #1817: [Feature Request] Trigger documentation updates when a new version is released
This a cross referenced issue for the pester/docs site https://github.com/pester/docs/issues/84

> CI/CD
> A solution was discussed earlier, something like:
> 
> pester/pester repo creates a new release
> pester/pester release triggers a github action in the pester/docs repo
> that pester/docs action uses the generate script to:
> regenerate the mdx files using LATEST pester version
> create a PR with the mdx file changes
> This way everything stays as it is now with wesbite-previews etc. and.. no need to move pester-docs-related logic into the pester/pester repo.

### PR DESCRIPTION
**Pull request by glennsarti**
Fixes #1817

This commit extends the release script to optionally trigger a GitHub Action
Workflow to update the Pester documentation site.  Note that this optional for
now until the required Github credentials are created and injected into the
release CI system (TeamCity?).  This commit also depends on the Docs GitHub
repository to have the correct Workflow in place.

See https://docs.github.com/rest/reference/actions#create-a-workflow-dispatch-event
for more information

### CODE REVIEW

**Comment by glennsarti** [comment]
This is a cross repository feature request from https://github.com/pester/docs/issues/84

**Comment by glennsarti** [comment]
This is dependant on https://github.com/pester/docs/pull/87 being merged first before it can be taken advantage of, however these changes themselves could be merged as the new behaviour is opt-in.

File: publish/release.ps1 [hunk 1]
```diff
@@ -132,6 +134,27 @@ Publish-Module -Path $psGalleryDir -NuGetApiKey $PsGalleryApiKey -Verbose -Force
...
+            'Accept' = 'application/vnd.github.v3+json'
+            'Authorization' = "Basic $encodedCreds"
+        }
+        $Body = @{
+            'ref' = 'master'
+            'inputs' = @{
+                'pester-version' = $Version
+            }
+        }
+        # There can be caching issues between the time a packaged is pushed to the PS Gallery, and when it becomes searchable. Hopefully waiting one
```
Comment (nohwnd) [inline publish/release.ps1:152]:
```suggestion
        # There can be caching issues between the time a package is pushed to the PS Gallery, and when it becomes searchable. Hopefully waiting one
```

File: publish/release.ps1 [hunk 2]
```diff
@@ -132,6 +134,27 @@ Publish-Module -Path $psGalleryDir -NuGetApiKey $PsGalleryApiKey -Verbose -Force
...
+        # minute gives the Gallery enough time to process the new package and make it available
+        Start-Sleep -Seconds 60
```
Comment (nohwnd) [inline publish/release.ps1:154]:
```suggestion
        Write-Host "Waiting 1 minute till package is on PSGallery"
        Start-Sleep -Seconds 60
```
Gimme some message so I know it's not stuck. 

Maybe polling using HEAD request after 20s on PSGallery and then every second till the resource is available would be better.

File: publish/release.ps1 [hunk 3]
```diff
@@ -132,6 +134,27 @@ Publish-Module -Path $psGalleryDir -NuGetApiKey $PsGalleryApiKey -Verbose -Force
...
+        # See https://docs.github.com/rest/reference/actions#create-a-workflow-dispatch-event for more information
+        Invoke-WebRequest -Uri 'https://api.github.com/repos/pester/docs/actions/workflows/update-from-pester.yml/dispatches' -Method POST -Headers $Headers -Body (ConvertTo-JSON -InputObject $Body)
+    }
 }
 else {
     Write-Host "This is pre-release $version, not pushing to Nuget and Chocolatey."
```
Comment (nohwnd) [inline publish/release.ps1:160]:
```suggestion
    Write-Host "This is pre-release $version, not pushing to Nuget and Chocolatey, and not updating docs."
```

**Comment by nohwnd** [comment]
Glenn thanks for this, but triggering it manually is easier for me than providing the token on build. As we discussed on the docs repo. I will close this without merge.

**Comment by glennsarti** [comment]
All good @nohwnd .  Thanks!

**Pull request closed without merging**